File: Sender.pas
var
  NowTesting, HostName, Message, Suffix: string;
  Port: integer;
  DT: DateTime;
  IP: System.Net.IPAddress;
  EP: System.Net.IPEndPoint;
  NS: System.Net.Sockets.NetworkStream;
  TCPClient: System.Net.Sockets.TcpClient;
  UDPClient: System.Net.Sockets.UdpClient;
  Buffer: array of byte;
  
begin
  Write('Имя хоста: ');
  Readln(HostName);
  IP := System.Net.Dns.GetHostAddresses(HostName)[0];
  Write('Порт: ');
  Readln(Port);
  EP := new System.Net.IPEndPoint(IP, Port);
  Write('Протокол: ');
  Readln(NowTesting);
  if (NowTesting.ToUpper = 'TCP') then
  begin
    TCPClient.Connect(EP);
    NS := TCPClient.GetStream();
    Writeln('Подключился! Ожидаю ваших сообщений...');
    while true do
    begin
      if NS.CanWrite then
      begin
      Suffix := '';
      Readln(Message);
      Buffer := System.Text.Encoding.ASCII.GetBytes(Message);
      try
        NS.Write(Buffer, 0, Buffer.Length);
      except
       on ex: Exception do
        Suffix := ex.Message;
      end;
      if (Suffix = '') then Suffix := '(Ошибки не произошло!)';
      DT := DateTime.Now;
      Writeln('<', DT.Hour, ':', DT.Minute, ':', DT.Second, '> ', Message, ' ', Suffix);
    end;
    End;
  end;
  if (NowTesting.ToUpper = 'UDP') then
  BEGIN
    UDPClient.Connect(EP);
    Writeln('Подключился! Ожидаю ваших сообщений...');
    while true do
    begin
      Suffix := '';
      Readln(Message);
      Buffer := Encoding.ASCII.GetBytes(Message);
      try
        UDPClient.Send(Buffer, Buffer.Length);
      except
      on ex: Exception do
        Suffix := ex.Message;
      end;
      if (Suffix = '') then Suffix := 'Операция прошла успешно!';
      Writeln('<', DT.Hour, ':', DT.Minute, ':', DT.Second, '> ', Message, ' ', Suffix);
    end;
  End;
end.

File: Sender.cs
using System;
using System.Net;
using System.Net.Sockets;

namespace Sender
{
    class Program
    {
        static void Main(string[] args)
        {
            DateTime DT = DateTime.Now;
            Console.Write("Имя хоста: ");
            string HostName = Console.ReadLine();
            System.Net.IPAddress IP = System.Net.Dns.GetHostAddresses(HostName)[0];
            Console.Write("Порт: ");
            int Port = Convert.ToInt32(Console.ReadLine());
            IPEndPoint EP = new IPEndPoint(IP, Port);
            Console.Write("Протокол: ");
            string NowTesting = Console.ReadLine();
            string Message;
            string Suffix;
            if (NowTesting == "TCP")
            {
                TcpClient Client = new TcpClient();
                Client.Connect(EP);
                NetworkStream NS = Client.GetStream();
                Console.WriteLine("Подключился! Ожидаю ваших сообщений...");
                while (true)
                {
                    if (NS.CanWrite)
                    {
                        Message = Console.ReadLine();
                        byte[] Buffer = System.Text.Encoding.ASCII.GetBytes(Message);
                        NS.Write(Buffer, 0, Buffer.Length);
                        Console.WriteLine($"<{DT.Hour}:{DT.Minute}:{DT.Second}>: {Message}");
                    }

                }
            }
            if (NowTesting == "UDP")
            {
                UdpClient Client = new UdpClient();
                Client.Connect(EP);
                Console.WriteLine("Подключился! Ожидаю ваших сообщений...");
                while (true) 
                {
                    Suffix = "";
                    Message = Console.ReadLine();
                    Byte[] Buffer = System.Text.Encoding.ASCII.GetBytes(Message);
                    try
                    {
                        Client.Send(Buffer, Buffer.Length);
                    }
                    catch (Exception e)
                    {
                        Suffix = e.ToString();
                    }
                    if (Suffix == "")
                    {
                        Suffix = "Операция прошла успешно!";
                    }
                    Console.WriteLine($"<{DT.Hour}:{DT.Minute}:{DT.Second}>: {Message} ({Suffix})");
                }
                
            }
        }
    }
}
